pagination.go

```diff
@@ -102,3 +124,60 @@ func (c *Client) listHelper(ctx context.Context, pager PagedResponse, opts *List
 	opts.Pages = pages
 	return nil
 }
+
+// flattenQueryStruct flattens a structure into a Resty-compatible query param map.
+// Fields are mapped using the `query` struct tag.
+func flattenQueryStruct(val any) (map[string]string, error) {
+	result := make(map[string]string)
+
+	reflectVal := reflect.ValueOf(val)
+
+	// Deref pointer if necessary
+	if reflectVal.Kind() == reflect.Ptr {
+		reflectVal = reflect.Indirect(reflectVal)
+	}
+
+	valType := reflectVal.Type()
+
+	for i := 0; i < valType.NumField(); i++ {
+		currentField := valType.Field(i)
+
+		queryTag, ok := currentField.Tag.Lookup("query")
+		// Skip untagged fields
+		if !ok {
+			continue
+		}
+
+		valField := reflectVal.FieldByName(currentField.Name)
+		if !valField.IsValid() {
+			return nil, fmt.Errorf("invalid query param tag: %s", currentField.Name)
+		}
+
+		// Skip if it's a zero value
+		if valField.IsZero() {
+			continue
+		}
+
+		// Deref the pointer is necessary
+		if valField.Kind() == reflect.Ptr {
+			valField = reflect.Indirect(valField)
+		}
+
+		var resultField string
+
+		switch valField.Interface().(type) {
+		case string:
+			resultField = valField.String()
+		case int64, int:
+			resultField = strconv.FormatInt(valField.Int(), 10)
+		case bool:
+			resultField = strconv.FormatBool(valField.Bool())
+		default:
+			return nil, fmt.Errorf("unsupported query param type: %s", valField.Type().Name())
+		}
```

Comment:
Do we want support slice (array) in query parameters? Maybe not today?

Reply:
I don't think it's necessary in this case since query params generally aren't expected to be provided in list format 👍